### ISSUE TITLE
Use existing IAM role

### ISSUE THREAD

**Issue by prashant358** [daniel-cottone/serverless-es-logs#216]
Hi Daniel,

Can we attach existing IAM role to es-logs lambda function? 

Thanks.

**Comment by angeldimitrov** [comment]
+1 :) Currently the default Role is getting way to big and CF does want to roll out
`
An error occurred: IamRoleLambdaExecution - Maximum policy size of 10240 bytes exceeded for role appla-api-dev-eu-central-1-lambdaRole (Service: AmazonIdentityManagement; Status Code: 409; Error Code: LimitExceeded; Request ID: 1c9d3f11-b77a-4882-b506-4478714663ab).
`

**Comment by noanflaherty** [comment]
+1! This plugin looks so close to being perfect for our needs. However, our lambdas run within a VPC and so we get this error when trying to deploy:
```
An error occurred: EsLogsProcesserLambdaFunction - The provided execution role does not have permissions to call CreateNetworkInterface on EC2 (Service: AWSLambdaInternal; Status Code: 400; Error Code: InvalidParameterValueException;
```
We have a role set up with all the proper permissions. Being able to use this instead would allow us to use this plugin.

**Comment by daniel-cottone** [comment]
There's been enough interest in this that I think we should add this feature. Perhaps adding an option to use the existing role?

**Comment by noanflaherty** [comment]
@daniel-cottone using an existing role sounds ideal as it allows for maximum flexibility!

**Comment by daniel-cottone** [comment]
@prashant358 @noanflaherty This is fixed and released in `2.3.0`. A couple of notes:

- If you simply need the log processor lambda to live in a VPC, I've updated the default role it uses so that it can acquire an ENI. If you were seeing errors like `The provided execution role does not have permissions to call CreateNetworkInterface on EC2` you should no longer see that now.
- If you truly want to override the role it uses and control that, you now can with the `useDefaultRole` option. However- you will need to add the specific permissions the lambda needs to write to elasticsearch yourself to your role.

In most situations you'll probably just want to let the plugin handle it, now that it can work in a VPC.

**Comment by noanflaherty** [comment]
@daniel-cottone thanks so much for turning this around so quickly! I'm excited to give it a try!  Will report back with how it works for us.
